### PR TITLE
901 report time split remaining

### LINKED ISSUE #901: Report shows "weeks remaining" of part-time or full-time splits
**Note**: branch off work here https://github.com/ctoec/ecis-experimental/pull/1020 ( includes https://github.com/ctoec/ecis-experimental/tree/895-new and https://github.com/ctoec/ecis-experimental/tree/895-reports-record-funding-weeks) if you begin this work before PR is merged 

**Prerequisites:** #895, #938 

**User story:** As a provider, I want to see how many weeks of full-time care are left on my contract in order to ensure I'm meeting my obligations.

**AC:**

* Using all submitted reports for the current fiscal year (July–June), determine the number of weeks that have already been claimed as full- or part-time (whichever is being asked about on the report).
* Display in the gray, italicized text as shown in the designs.
* As the text field is updated, update the weeks remaining count.
* Prevent a number from being entered into the text field that is greater than the number of weeks remaining.

## When the count is 0

Screen Shot 2020-03-19 at 10 28 59 AM

## After the user has edited the count

Screen Shot 2020-03-19 at 12 29 18 PM

### LINKED ISSUE #972: Create backend integration tests for Report PUT
Write an integration test that hits the `api/organizations/[orgId]/Reports/[reportId]` endpoint with a PUT body of report object.

Assert the response has the appropriate structure:
* `organization` not null
* `organization.sites` not null
* `organization.sites.organization` null
* `organization.sites.enrollments` null
* `organization.fundingsSpaces.fundings` null
* `enrollments.child` not null
* `enrollments.child.organization` null
* `enrollments.child.family` null
* `enrollments.child.enrollments` null
* `enrollments.site` null
* `enrollments.fundings` not null
* `enrollments.fundings.firstReportingPeriod` null
* `enrollments.fundings.lastReportingPeriod` null
* `enrollments.fundings.enrollment` null
* `enrollments.fundings.fundingSpace` null
* `enrollments.pastEnrollments` null

Assert no validation errors

### PR DESCRIPTION
**Pull request by cailyncodes**
Should close #901 and close #972

### CODE REVIEW

File: src/Hedwig/ClientApp/src/containers/Reports/ReportDetail/ReportSubmitForm.tsx [hunk 1]
```diff
@@ -162,75 +168,123 @@ export default function ReportSubmitForm({
...
 			  )
 	);
 
 	const splitTimeUtilizationQuestion = (fundingSpace: FundingSpace) => {
 		const timeSplit = fundingSpace.timeSplit;
 		if (!timeSplit) return;
-
+		const currentFiscalYearTimeSplitUtilizations = ((fundingSpace.timeSplitUtilizations ||
+			[]) as DeepNonUndefineable<FundingTimeSplitUtilization[]>)
+			.filter(util => util.reportId !== report.id)
```
Comment (jlhogan) [inline src/Hedwig/ClientApp/src/containers/Reports/ReportDetail/ReportSubmitForm.tsx:205]: pull this out into fundingspace util function?

File: src/Hedwig/ClientApp/src/containers/Reports/ReportDetail/ReportSubmitForm.tsx [hunk 2]
```diff
@@ -162,75 +168,123 @@ export default function ReportSubmitForm({
...
+				moment(util.reportingPeriod.periodEnd).isBefore(report.reportingPeriod.periodEnd)
+			)
+			.filter(util => !!util.report.submittedAt);
 		const lesserTime =
 			timeSplit.fullTimeWeeks < timeSplit.partTimeWeeks ? FundingTime.Full : FundingTime.Part;
 		const labelText = `${prettyAge(
 			fundingSpace.ageGroup
 		)} services were provided ${prettyFundingTime(lesserTime)}`;
 
+		const fiscalYearUsedWeeks = currentFiscalYearTimeSplitUtilizations.reduce((acc, util) => {
```
Comment (jlhogan) [inline src/Hedwig/ClientApp/src/containers/Reports/ReportDetail/ReportSubmitForm.tsx:220]: another util?

File: src/Hedwig/Repositories/ReportRepository.cs [hunk 3]
```diff
@@ -65,7 +65,11 @@ public async Task<CdcReport> GetReportForOrganizationAsync(int id, int orgId, st
 					.Include(report => report.Organization)
 						.ThenInclude(organization => organization.FundingSpaces)
 							.ThenInclude(fundingSpace => fundingSpace.TimeSplitUtilizations)
-								.ThenInclude(timeSplitUtilization => timeSplitUtilization.ReportingPeriod);
+								.ThenInclude(timeSplitUtilization => timeSplitUtilization.ReportingPeriod)
+					.Include(report => report.Organization)
+						.ThenInclude(organization => organization.FundingSpaces)
+							.ThenInclude(fundingSpace => fundingSpace.TimeSplitUtilizations)
+								.ThenInclude(util => util.Report);
```
Comment (jlhogan) [inline src/Hedwig/Repositories/ReportRepository.cs:72]: if you end up pulling the reports here on the timesplitutilizations anyway, is it useful to also have the hoisted up reporting period?

File: src/Hedwig/Models/Extensions/ReportExtensions.cs [hunk 4]
```diff
@@ -0,0 +1,18 @@
+using System.Linq;
+using System.Collections.Generic;
+
+namespace Hedwig.Models.Extensions
+{
+	public static class ReportExtensions
+	{
+		public static int GetWeeksUsedForFundingSpace(this IEnumerable<CdcReport> reports, FundingSpace fundingSpace)
```
Comment (jlhogan) [inline src/Hedwig/Models/Extensions/ReportExtensions.cs:8]: it seems unexpected that this logic would belong to a set of reports, but i'm not sure of an alternative that i like more. maybe would expect this to happen from the fundingspace entrypoint, where it would inspect it's collection of timeutilizations, and compare the reporting periods on those to a fiscal year parameter? but that relies on the assumption that if time split utilizations exist, they are associated with a submitted report. is that always true? 

tldr; this works but for some reason was not what i expected but not a blocker in any way

File: src/Hedwig/Validations/Attributes/Reports/Cdc/FundingTimeUtilizationsWeeksUsedAreLessThanReportingPeriodWeeks.cs [hunk 5]
```diff
@@ -0,0 +1,42 @@
...
+using Hedwig.Models.Extensions;
+
+namespace Hedwig.Validations.Attributes
+{
+	public class FundingTimeUtilizationsWeeksUsedAreLessThanReportingPeriodWeeks : ValidationAttribute
+	{
+		protected override ValidationResult IsValid(object value, ValidationContext validationContext)
+		{
+			var report = validationContext.ObjectInstance as CdcReport;
+			var timeSplitUtilizations = value as ICollection<FundingTimeSplitUtilization> ?? new List<FundingTimeSplitUtilization> { };
```
Comment (jlhogan) [inline src/Hedwig/Validations/Attributes/Reports/Cdc/FundingTimeUtilizationsWeeksUsedAreLessThanReportingPeriodWeeks.cs:16]: do these need to be grabbed from repository if they're not on the object? edge case if at all

File: src/Hedwig/ClientApp/src/assets/styles/_general.scss [hunk 6]
```diff
@@ -197,3 +197,7 @@ a.usa-button {
 		margin-left: -1.55rem;
 	}
 }
+
+.usa-form--full-width {
```
Comment (mmazanec22) [inline src/Hedwig/ClientApp/src/assets/styles/_general.scss:201]: We maybe shouldn't be trying to override the USWDS default?  https://designsystem.digital.gov/components/form-templates/

File: src/Hedwig/ClientApp/src/components/TextInput/TextInput.tsx [hunk 7]
```diff
@@ -1,4 +1,4 @@
-import React from 'react';
+import React, { HtmlHTMLAttributes } from 'react';
```
Comment (mmazanec22) [inline src/Hedwig/ClientApp/src/components/TextInput/TextInput.tsx:1]: Looks like this is added but not used

File: src/Hedwig/Validations/Attributes/Reports/Cdc/FundingTimeUtilizationsWeeksUsedAreLessThanReportingPeriodWeeks.cs (hunk 5 above)
Comment (cailyncodes) [reply src/Hedwig/Validations/Attributes/Reports/Cdc/FundingTimeUtilizationsWeeksUsedAreLessThanReportingPeriodWeeks.cs:16]: I don't think so. If they are not on the report PUT object, that means we a) don't have utils for this report/any funding spaces that have splits, or b) we forgot to include them in initial report GET, and because then a funding space in this report is split, if we fetch them we won't have the newest entry and will give bad signal on validity.

File: src/Hedwig/ClientApp/src/assets/styles/_general.scss (hunk 6 above)
Comment (cailyncodes) [reply src/Hedwig/ClientApp/src/assets/styles/_general.scss:201]: Without that, we get this
image
